### ISSUE TITLE
Experimental : The Leaderboard

### ISSUE THREAD

**Issue by Shrijul-Venkatesh-4gd** [Shrijul-Venkatesh-4gd/autoresearch#2]
I have now added the ability to auto-update a unified leaderboard for each experiment run. This is what the second experiment looked like :  

| run_id                                                         |  timestamp                 |  val_bpb  |  train_loss_final |  training_seconds |  total_seconds |  peak_vram_mb |  tok_per_sec |  mfu_percent |  total_tokens_M |  num_steps |  num_params_M |  depth |  sequence_len |  device_batch_size |  total_batch_size |  window_pattern |  config_name |  notes |
| -------------------------------------------------------------- | -------------------------- | --------- | ----------------- | ----------------- | -------------- | ------------- | ------------ | ------------ | --------------- | ---------- | ------------- | ------ | ------------- | ------------------ | ----------------- | --------------- | ------------ | ------ |
| 2026-03-14T14:44:42+00:00_994a71b6-37f7-4b05-9931-b0cfb309af14 |  2026-03-14T14:44:42+00:00 |  1.354374 |          3.785250 |             300.0 |          312.3 |         258.9 |       193914 |         0.65 |            51.1 |       3116 |          11.5 |      4 |           512 |                  4 |             16384 |  L              |              |        |

Comparing these results to what we saw on #1, we see that the second experiment was slightly better when compared based on `val_bpb` as this improved from 1.357812 to 1.354374. The training loss also dropped as well with another very notable improvement being the throughput improving from 78.9k tok/sec to 193.9k tok/sec. However since there was no change made to `train.py` (there was no agent tuning the hyperparams at any stage), most of these improvements can be attributed to pure luck. Hence, we need multiple training loops to be performed in order for us to even arrive at a solid conclusion.

**Comment by Shrijul-Venkatesh-4gd** [comment]
Since the reduction of the training params did not peak my RTX 3050, I proceeded to bump up the values by a bit to see improvement in the `val_bpb` score. These are the changes that were made : 
```
MAX_SEQ_LEN = 512
DEPTH = 6
DEVICE_BATCH_SIZE = 8
TOTAL_BATCH_SIZE = 2**15
```

And this is what the leaderboard looks like now. `val_bpb` dropped to 1.327390 which was the biggest drop seen so far, and the model doubled to 26.3M params. Due to this the model saw much less data in the same wall-clock budget as the data throughput dropped significantly. The interesting part is that the final training loss, 3.795335, is only a little better than the previous runs, but the validation score improved much more noticeably. That is a good sign, because it suggests the larger model is generalizing better on the held-out validation text rather than just fitting the training batches differently

| run_id                                                         |  timestamp                 |  val_bpb  |  train_loss_final |  training_seconds |  total_seconds |  peak_vram_mb |  tok_per_sec |  mfu_percent |  total_tokens_M |  num_steps |  num_params_M |  depth |  sequence_len |  device_batch_size |  total_batch_size |  window_pattern |  config_name |  notes |
| -------------------------------------------------------------- | -------------------------- | --------- | ----------------- | ----------------- | -------------- | ------------- | ------------ | ------------ | --------------- | ---------- | ------------- | ------ | ------------- | ------------------ | ----------------- | --------------- | ------------ | ------ |
| 2026-03-14T14:44:42+00:00_994a71b6-37f7-4b05-9931-b0cfb309af14 |  2026-03-14T14:44:42+00:00 |  1.354374 |          3.785250 |             300.0 |          312.3 |         258.9 |       193914 |         0.65 |            51.1 |       3116 |          11.5 |      4 |           512 |                  4 |             16384 |  L              |              |        |
| 2026-03-14T15:01:21+00:00_1ebdc4f1-4fa5-4adc-80ac-00c6ad8aa3b5 | 2026-03-14T15:01:21+00:00  | 1.327390  | 3.795335          | 300.4             | 334.5          | 752.3         | 69641        | 0.93         | 29.0            | 884        | 26.3          | 6      | 512           | 8                  | 32768             | L               |              |        |

<img width="1857" height="894" alt="Image" src="https://github.com/user-attachments/assets/b76a8e28-dc1c-47ca-ab5b-501a32815083" />

<img width="1857" height="894" alt="Image" src="https://github.com/user-attachments/assets/57311a45-4f12-41e3-85a8-270dd4d9cf59" />

**Comment by Shrijul-Venkatesh-4gd** [comment]
I have finally updated the values to the following config : 
```
MAX_SEQ_LEN = 512
DEPTH = 8
DEVICE_BATCH_SIZE = 8
TOTAL_BATCH_SIZE = 2**16
```

If the general trend followed, then one would expect that this configuration would be significantly better than the previous runs. However, this was far from reality. The larger 8-layer configuration ran successfully, and it still fit comfortably in memory, reaching about 1.29 GB of peak VRAM, but it turned out to be slower and less effective than my earlier 6-layer setup. This model had 50.3 million parameters and trained for the full 5-minute budget, but because each step was much heavier it only completed 243 steps and processed 15.9 million tokens. The final validation score was `val_bpb = 1.335136`, which is better than the smaller 4-layer runs but worse than the best 6-layer run at `1.327390`. That suggests I pushed the model size past the sweet spot for my hardware: even though the GPU handled it, the model became too slow within the fixed time budget to produce the best overall result.

### This is my machine right now 😭😭😭😭

<img width="1857" height="894" alt="Image" src="https://github.com/user-attachments/assets/f8486031-e03f-40d5-a937-d2ad9b7a9068" />

<img width="1857" height="894" alt="Image" src="https://github.com/user-attachments/assets/dae87311-4106-4cdd-9a9d-b11d70c4e8e5" />
